gr-digital/lib/meas_evm_cc_impl.cc

```diff
@@ -0,0 +1,78 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2020 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "meas_evm_cc_impl.h"
+#include <gnuradio/io_signature.h>
+
+namespace gr {
+namespace digital {
+
+meas_evm_cc::sptr meas_evm_cc::make(constellation_sptr cons, evm_measurement_t meas_type)
+{
+    return gnuradio::get_initial_sptr(new meas_evm_cc_impl(cons, meas_type));
+}
+
+meas_evm_cc_impl::meas_evm_cc_impl(constellation_sptr cons, evm_measurement_t meas_type)
+    : gr::sync_block("meas_evm_cc",
+                     gr::io_signature::make(1, 1, sizeof(gr_complex)),
+                     gr::io_signature::make(1, 1, sizeof(float))),
+      d_cons(cons),
+      d_meas_type(meas_type)
+{
+
+    d_cons_points = d_cons->points();
+
+    double sum = 0.0;
+    int N = d_cons_points.size();
+    for (int i = 0; i < N; i++) {
+        sum += pow(real(d_cons_points[i]), 2.0) + pow(imag(d_cons_points[i]), 2.0);
```

Comment:
I was about to write "um, `pow(x, 2.0)` sounds like a very expensive way of writing `x*x`: Modern GCC [doesn't seem to care](https://gcc.godbolt.org/z/Ranv-p). The only difference is the order of additions?! nice.

tbh, this is again not really asking for improvement, but `std::norm` would keep the calculation float for as long as it's not detrimental, so should be faster. Also, VOLK, maybe?

Reply:
But really, this is a one-time thing in the constructor; couldn't care less.